Sources/NIO/EventLoop.swift

```diff
@@ -464,16 +468,17 @@ internal final class SelectableEventLoop: EventLoop {
     public func run() throws {
         precondition(self.inEventLoop, "tried to run the EventLoop on the wrong thread.")
         defer {
-            var tasksCopy = ContiguousArray<ScheduledTask>()
-
+            var scheduledTasksCopy = ContiguousArray<ScheduledTask>()
             tasksLock.lock()
+            // reserve the correct capacity so we don't need to realloc later on.
+            scheduledTasksCopy.reserveCapacity(scheduledTasks.count)
```

Comment:
shouldn't this happen before we take the lock?

Reply:
We need the lock because we access scheduledTasks.count

Reply:
thanks, makes sense